libclambcc/ClamBCLogicalCompiler/ClamBCLogicalCompiler.cpp

```diff
@@ -934,6 +943,27 @@ class LogicalCompiler
             }
 
             Instruction *pInst = llvm::cast<Instruction>(I);
+#if 0
+            /*Look through all operands of the instruction and add the
+             * constants to the logical map, so that we won't fail to create
+             * the siganture if O3 changes a logical expression to a constant
+             * at compile time.
+             */
+            for (size_t i = 0; i < pInst->getNumOperands(); i++){
+                if (ConstantInt * pci = llvm::dyn_cast<ConstantInt>(pInst->getOperand(i))){
+                    if (pci->isOne()){
+                        LogicalNode * ln = LogicalNode::getTrue(allNodes) ;
+                        Map[pci] = ln;
+                    } else if (pci->isZero()){
+                        LogicalNode * ln = LogicalNode::getTrue(allNodes) ;
+                        Map[pci] = ln;
+                    }
+                }
+            }
+#else
+            DEBUGERR << "TEMPORARILY REMOVED TO FIGURE THIS OUT" << "<END>\n";
```

Comment:
No, I will remove the print before I merge.  The other code block I am leaving for now, because this file will likely need more editing during testing, and I am not sure if I will revisit that idea.